corehq/apps/es/forms.py

```diff
@@ -92,42 +92,46 @@ def _from_dict(cls, xform):
         from corehq.apps.receiverwrapper.util import get_app_version_info
         from corehq.pillows.utils import format_form_meta_for_es, get_user_type
         from corehq.pillows.xform import is_valid_date
+        doc_ret = copy(xform)
+        form = copy(doc_ret['form'])
+        form_meta = copy(form.get('meta', None))
```

Comment:
Assigning the copies immediately here would make it easier to see how they relate to `doc_ret` eventually.
```suggestion
        form = doc_ret['form'] = copy(doc_ret['form'])
        form_meta = form['meta'] = copy(form.get('meta', None))
```